### DISCUSSION TITLE
How to split by newline?

### DISCUSSION THREAD

**Discussion by burkostya** [MontFerret/ferret#643] [Q&A] [answered]
How to split string by `\n`?

I tried next query
```FQL
LET doc = DOCUMENT('https://github.com/events', {driver: 'cdp'})
LET list = ELEMENT(doc, '.footer')[0]
LET str = INNER_TEXT(list) // str contains "© 2021 GitHub, Inc.\nTerms\nPrivacy\nSecurity\nStatus\nDocs\nContact GitHub\nPricing\nAPI\nTraining\nBlog\nAbout"
RETURN SPLIT(str, '\\n')
```

But it returns not splitted string
```json
[
    "© 2021 GitHub, Inc.\nTerms\nPrivacy\nSecurity\nStatus\nDocs\nContact GitHub\nPricing\nAPI\nTraining\nBlog\nAbout"
]
```

**Comment by ziflex** [comment, marked as answer]
Hey, did you try without escaping ``\``? 
Just ```RETURN SPLIT(str, '\n')```

**Reply by burkostya** [reply]
Yes, result the same.

**Reply by ziflex** [reply]
What version of Ferret are you using?

**Reply by ziflex** [reply]
Ok, I can see the issue now.

**Reply by bundleman** [reply]
@burkostya Hi! Try using other quota - "", and only one slash. I mean - ```
LET str = "© 2021 GitHub, Inc.\nTerms\nPrivacy\nSecurity\nStatus\nDocs\nContact GitHub\nPricing\nAPI\nTraining\nBlog\nAbout"
RETURN SPLIT(str,  "\n")```

**Reply by burkostya** [reply]
The same result in ferret from master.
@bundleman Your script is working. When variable `str` initialized with literal string (like in your script) then `SPLIT` splits the string as expected. But when `str` initialized with result of `INNER_TEXT` (like in the topic script) then `SPLIT` does not split the string.

**Reply by ziflex** [reply]
@burkostya we are working in this one. It will be fixed in the upcoming 0.16.0 release.

**Reply by ziflex** [reply]
Oh, hey. Forgot to let you know that it's fixed in v0.16.

**Reply by gonssal** [reply]
I'm using v0.16.3 and it still doesn't seem to work. It's not only `SPLIT()`, but also other functions like `SUBSTITUTE()`.

**Reply by ziflex** [reply]
> I'm using v0.16.3 and it still doesn't seem to work. It's not only `SPLIT()`, but also other functions like `SUBSTITUTE()`.

Could you provide an example?

**Reply by gonssal** [reply]
> > I'm using v0.16.3 and it still doesn't seem to work. It's not only `SPLIT()`, but also other functions like `SUBSTITUTE()`.
> 
> Could you provide an example?

Sure:

```
LET propertyUrl = 'https://www.and.immo/venda/pis-escaldes-engordany-suite/522-ref1659'
LET propDoc = DOCUMENT(propertyUrl)
LET property = {
	Bedrooms: SUBSTITUTE(SPLIT(TRIM(INNER_TEXT(propDoc, '.text_home .caracteristiquesfitxainterior')), '                                    ')[1], "\\n", ""),
}
RETURN property
```

Also, the example code in `examples/split-new-line.fql` gives me the following error: `cdp.DOM: ResolveNode: rpc error: No node with given id found (code = -32000): ELEMENT(doc,'.footer')[0] at 2:11`. If I change it to this...
```
LET doc = DOCUMENT('https://github.com/events', {driver: 'cdp'})
LET str = INNER_TEXT(doc, '.footer')
RETURN SPLIT(str, "\\n")
```
... then it works, but still doesn't replace the `\n`s.

Because running `ferret --version` doesn't return anything I'm doubtful about the version, but I'm pretty sure it's 0.16.3 because in `$GOPATH/pkg/mod/github.com/!mont!ferret/` I have `cli@v1.5.0` and `ferret@v0.16.3` and in my project's go.mod it's also v0.16.3. It would be nice if `--version` returned both the cli and the library versions.

**Reply by ziflex** [reply]
Try just ``SPLIT(str, "\n")``

**Reply by ziflex** [reply]
Unfortunately, when you install ``ferret`` using ``go get`` or ``go install`` go does not set the version number. For that, you would need to download a binary.

**Reply by gonssal** [reply]
It works now, thank you very much.
